### PR TITLE
[Ticket/14131] Completely overhaul all colors in Colours.css

### PR DESCRIPTION
**Pull request by hanakin**
Completely overhaul all colors in Colours.css to be consistent and standardized. Match all colors to a color styleguide using set color variables.

Tracker: https://tracker.phpbb.com/browse/PHPBB3-14131

PHPBB3-14131

### CODE REVIEW

**Comment by hanakin** [comment]
!set Styles

**Comment by Zoddo** [comment]
See https://github.com/phpbb/phpbb/pull/3866#issuecomment-136151656 :)

**Comment by hanakin** [comment]
Ready for review!

**Comment by Nicofuma** [comment]
I guess that not every browser supports this feature, right? What about the one we officially support?

**Comment by Nicofuma** [comment]
Well actually, I don't really understand what you are doing... just cleaning up the formatting and some weird color definitions?

**Comment by hanakin** [comment]
Nothing is supported by a browser those are just the variables I used when I fixed all the incorrect colors in the theme which is the main purpose of this. I removed all the slight variances and replaced all the incorrect colors with the proper ones based on the standardized guide in the comments. I left the comments in for documentation purposes on what colors to use in what situations. Overall reducing the total number of colors in half

**Comment by Nicofuma** [comment]
ok, thanks.

By the way, you need to rebase (it conflicts with another of your PRs)

**Comment by hanakin** [comment]
yeah working that now there is alot of manual review lol

**Comment by Nicofuma** [comment]
@cyberalien Again, could you have a look please? (Yes, there is a lot of styling PRs these days :-))

File: phpBB/styles/prosilver/theme/colours.css [hunk 1]
```diff
@@ -1,24 +1,106 @@
...
+
+html,
+body {
+	color: #5C758C;
 	background-color: #F5F7FA;
 }
 
-h1 {
-	color: #FFFFFF;
+h1 {
```
Comment (marc1706) [inline phpBB/styles/prosilver/theme/colours.css]: Trailing whitespace should be removed

**Comment by marc1706** [comment]
Your PR seems to contain a lot of trailing whitespace. They're nicely highlighted by github, too.

**Comment by hanakin** [comment]
@marc1706 yeah have not really done much work on as every merge affects it drastically it going to have to go as the last merge anyway. So its best to fix it all during the long dynamic rebase

**Comment by hanakin** [comment]
!set WIP

**Comment by cyberalien** [comment]
Sorry for delay. Color changes look good.

**Comment by Zoddo** [comment]
@hanakin : You need to add `PHPBB3-14131` at the end of each commit message for the tracker.

**Comment by hanakin** [comment]
ready for review @cyberalien

**Comment by hanakin** [comment]
!unset WIP

**Comment by hanakin** [comment]
ready for review @prototech should not be any BC issues just helping to standardize colours.

**Comment by VSEphpbb** [comment]
Umm,forum background color is awful. Please revert that. There's just a weird wash of blue-i-ness to this that makes phpBB uglier.

**Comment by hanakin** [comment]
@VSEphpbb used bg instead of 2 how about now?

**Comment by VSEphpbb** [comment]
Actually, I was talking about the body/html background.

**Comment by hanakin** [comment]
@VSEphpbb

**Comment by marc1706** [comment]
The post text in the editor is also highlighted in the same color as the outside of the text box border. This looks kind of weird.

**Comment by hanakin** [comment]
@marc1706 ill have a look when i get home later tonight

**Comment by hanakin** [comment]
@marc1706 try now switched it back to the darker blue

**Comment by VSEphpbb** [comment]
Text in post editor should be black. If that's what's being discussed here.

**Comment by hanakin** [comment]
hmm can make it default color but its always been dark blue

**Comment by VSEphpbb** [comment]
It's always been black. technically, `#333333`
https://github.com/phpbb/phpbb/blob/3.1.x/phpBB/styles/prosilver/theme/colours.css#L973-L987

**Comment by hanakin** [comment]
hmm guess the specificity was coded backwords originally not sure why any of the inputs was blue to begin with possibly easier to leave it as all black and remove/deprecate that line.

**Comment by VSEphpbb** [comment]
I have to say I'm still not a fan of some of the color changes, or why they are even needed. To my eyes, this PR is just hurting things with the color changes/choices.

That said, still many inconsistencies with font colors, leading to them harder to read. Font colors are too washed out, and sometimes a weird off blue/green when they should be black.

THIS PR:
screen shot 2015-11-23 at 7 35 48 pm
screen shot 2015-11-23 at 7 35 36 pm

Original (better):
screen shot 2015-11-23 at 7 36 41 pm
screen shot 2015-11-23 at 7 36 53 pm

**Comment by hanakin** [comment]
the problem was the overuse of colors. there were an extreme amount of unique colors being used in the theme making it harder to manage and properly fix things. This is a perfect example of poor design/color usage from the start as you stated should probably be black and I agree but the original was not black. it was a randomly chose tint of blue which had no ties to any of the primary or what should have been the primary colors. A primary color scheme was never chosen, instead the individual colors were chosen on each mockup and thus we ended up with hundreds of unique colors which is a nightmare for web/design. It makes it harder for theme authors to modify or extension authors to add too if they can not easily see what color to use.

There was probably 5-6 different colors for basic text/headers alone. which should not be the case you should only have 1 and possibly a lighter shade of it for subs.

The methods that I used for replace was not optimal for fixing every individual case and but it was the easiest way to standardize the majority.

here is with the secondary dark color to attempt keeping some of the original blue
screenshot 2015-11-24 07 48 30

**Comment by VSEphpbb** [comment]
Still, I don't see any value to messing with Prosilver, when these efforts could be better spent on work for an all new design/theme. This is probably going to make the 3.1 to 3.2 transition for style authors another nightmare (and the 3.0 to 3.2 even more so).

**Comment by hanakin** [comment]
it should have little if any impact on authors current themes as no BC is broken all thats changes is the default colors for more consistancy. As for the why yes a new theme is needed and probably better spent working on it but it will be some time until a new theme is finished can be included with the core and completely replaces prosilver to the point where we can not support it anymore given the extent of extensions and styles already based of it.

**Comment by VSEphpbb** [comment]
Maybe the bug fixes should be separated from this conceptual change into two PRs (bugs such as message editor text currently being blue)

**Comment by hanakin** [comment]
I agree that fixes should be separate and for the most part this should adhere to it. the only deviations is either formating to make it easier to read which can be undone or implemented in 3.2 ie the icon colors.

**Pull request closed without merging**